### DISCUSSION TITLE
session.exec does not support delete statements

### DISCUSSION THREAD

**Discussion by McEazy2700** [fastapi/sqlmodel#821] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import (col, delete)
from common.models import LibrarySoundNFTLink


           
 delete_stmt = delete(LibrarySoundNFTLink).where(
   col(LibrarySoundNFTLink.library_user_address) == address
 )
 session.exec(delete_stmt)
```

### Description

session.exec complains when  executing a delete operation with the following error
```
Pyright: No overloads for "exec" match the provided arguments [reportCallIssue]
Pyright: Argument of type "Delete" cannot be assigned to parameter "statement" of type "SelectOfScalar[_TSelectParam@exec]" in function "exec"
  "Delete" is incompatible with "SelectOfScalar[_TSelectParam@exec]" [reportArgumentType]
```

I hear I should use `session.execute` to execute bulk delete operations, but session.execute will soon be deprecated (the warning keeps buging me)

### Operating System

Linux

### SQLModel Version

0.0.14

### Python Version

3.12

**Comment by jackbravo** [comment]
I get the same, this seems like an issue instead of a question :-p

**Reply by YoraiLevi** [reply]
any directions?

**Reply by monchin** [reply]
just use `execute` and ignore the deprecated warning...

**Reply by YaoJusheng** [reply]
Has the problem been solved? I also encountered the same problem.

I used `session.delete` instead.

**Comment by N-Demir** [comment]
Yeah what's going on here?

**Comment by Burtsev-Alexey** [comment]
Still doesn't work

**Reply by dev-prgmer** [reply, marked as answer]
Appears to have a [issue 909](https://github.com/fastapi/sqlmodel/issues/909) ticket open.
